control-plane/grpc/src/operations/volume/client.rs

```diff
@@ -0,0 +1,203 @@
+use crate::{
+    common::VolumeFilter,
+    grpc_opts::{Client, Context},
+    operations::volume::traits::{
+        CreateVolumeInfo, DestroyVolumeInfo, PublishVolumeInfo, SetVolumeReplicaInfo,
+        ShareVolumeInfo, UnpublishVolumeInfo, UnshareVolumeInfo, VolumeOperations,
+    },
+    volume::{
+        create_volume_reply, get_volumes_reply, get_volumes_request, publish_volume_reply,
+        set_volume_replica_reply, share_volume_reply, unpublish_volume_reply,
+        volume_grpc_client::VolumeGrpcClient, GetVolumesRequest, ProbeRequest,
+    },
+};
+use common_lib::{
+    mbus_api::{v0::Volumes, ReplyError, ResourceKind, TimeoutOptions},
+    types::v0::message_bus::{Filter, MessageIdVs, Volume},
+};
+use std::{convert::TryFrom, ops::Deref};
+use tonic::transport::{Channel, Uri};
+
+/// RPC Pool Client
+#[derive(Clone)]
+pub struct VolumeClient {
+    inner: Client<VolumeGrpcClient<Channel>>,
+}
+
+impl VolumeClient {
+    /// creates a new base tonic endpoint with the timeout options and the address
+    pub async fn new<O: Into<Option<TimeoutOptions>>>(addr: Uri, opts: O) -> Self {
+        let client = Client::new(addr, opts, VolumeGrpcClient::new).await;
+        Self { inner: client }
+    }
+}
+
+impl Deref for VolumeClient {
+    type Target = Client<VolumeGrpcClient<Channel>>;
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+/// Implement pool operations supported by the Pool RPC client.
+/// This converts the client side data into a RPC request.
+#[tonic::async_trait]
+impl VolumeOperations for VolumeClient {
+    async fn create(
+        &self,
+        create_volume_req: &dyn CreateVolumeInfo,
+        ctx: Option<Context>,
+    ) -> Result<Volume, ReplyError> {
+        let req = self.request(create_volume_req, ctx, MessageIdVs::CreateVolume);
+        let response = self.client().clone().create_volume(req).await?.into_inner();
```

Comment:
don't need to keep calling clone as client returns a cloned client already?

Reply:
Right. Removing them.